extension/src/openvic-extension/singletons/MenuSingleton.cpp

```diff
@@ -488,7 +489,28 @@ Dictionary MenuSingleton::get_province_info_from_index(int32_t index) const {
 	} else {
 		ret[province_info_rgo_employment_percentage_key] = (rgo.get_total_employees_count_cache() * fixed_point_t::_100() / max_employee_count).to_float_rounded();
 	}
+	if (rgo.is_valid()) {
 
+		String amount_of_employees_by_pop_type;
+		auto const& employee_count_per_type_cache = rgo.get_employee_count_per_type_cache();
+		for (PopType const& pop_type : *employee_count_per_type_cache.get_keys()) {
+			const pop_size_t employees_of_type = employee_count_per_type_cache[pop_type];
+			if (employees_of_type > 0) {
+				amount_of_employees_by_pop_type += Utilities::std_to_godot_string(pop_type.get_identifier().data()) + ": " +
+					Utilities::std_to_godot_string(std::to_string(employees_of_type)) + "\n";
+			}
+		}
+
+		ret[province_info_rgo_employment_tooltip_key] = tr("Current employees in RGO") + "\n" + "--------------\n" +
+			tr("Current employees") + ": " + static_cast<String>(ret[province_info_rgo_total_employees_key]) + "/" +
+			static_cast<String>(static_cast<Variant>(rgo.get_max_employee_count_cache())) + "\n" +
+			amount_of_employees_by_pop_type + tr("Base Workforce") + ": " +
+			static_cast<String>(static_cast<Variant>(rgo.get_production_type_nullable()->get_base_workforce_size())) + "\n" +
+			"From technology" + ":" + "\n" + "Province Size: x " +
+			static_cast<String>(static_cast<Variant>(rgo.get_size_multiplier().to_float()));
+	} else {
+		Logger::error("Rgo is invalid at", province->get_identifier());
```

Comment:
Don't log error if rgo isn't valid.
It's okay for the rgo to not be valid for water provinces.

If you find this confusing, I kinda agree. Take it up with hop.
He reasoned that this approach is better for memory management than using a pointer for rgo,

Reply:
it's all good that is a little weird but who am I to judge